transports/sftp/src/main/java/org/mule/transport/sftp/SftpReceiverRequesterUtil.java

```diff
@@ -213,7 +213,17 @@ public InputStream retrieveFile(String fileName, SftpNotifier notifier) throws E
         // This special InputStream closes the SftpClient when the stream is closed.
         // The stream will be materialized in a Message Dispatcher or Service
         // Component
-        return new SftpInputStream(client, fileInputStream, fileName, connector.isAutoDelete(), endpoint);
+        return new SftpInputStream(client, fileInputStream, fileName, determineAutoDelete(), endpoint);
+    }
+
+    private boolean determineAutoDelete()
+    {
+        Boolean autoDelete = Boolean.valueOf((String) endpoint.getProperty("autoDelete"));
```

Comment:
this is wrong. valueOf() will never return null. On the contrary, it will return false if the input is null.

The null check you actually have to do is against the endpoint property